### ISSUE TITLE
Events don't inherit "editable" properties from FullCalendar "editable" option

### ISSUE THREAD

**Issue by Cipscis** [fullcalendar/fullcalendar#7596]
### Reduced Test Case

https://codepen.io/Cipscis/pen/rNRqoXg

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

No connector (vanilla JS)

### Bug Description

When [FullCalendar's editable property](https://fullcalendar.io/docs/editable) is set to `true`, block events without an editable property of their own will inherit their behaviour from FullCalendar's property. In this way, they behave just like they were created with [their own editable property](https://fullcalendar.io/docs/event-object) set to the same value as the FullCalendar option.

However, when the event object appears (e.g. in callbacks from UI events), the `startEditable` and `durationEditable` properties will have the value `undefined` instead of a boolean value.

This is particularly unclear in TypeScript, as these properties are typed as required `boolean` properties on the `EventApi` interface, but in reality they aren't always present with a `boolean` value.

In my [minimal reproducible example](https://codepen.io/Cipscis/pen/rNRqoXg), a block event which only inherits `editable: true` via the FullCalendar property will have its `startEditable` property logged to the console when it is dragged to a different time, using the `eventDrop` callback.

The **expected behaviour** is that this property's `startEditable` and `durationEditable` values are both `true`, reflecting that the event's start time and durations are both editable.

The **actual behaviour** is that these properties' values are both `undefined`.

**Comment by acerix** [comment]
I'm not sure it's a bug that events do not inherit the `editable` property from the calendar, since it could be useful to know if an event has the `editable` property independent of the calendar, but it does seem like that type should also allow `undefined` so I believe it's a bug with that type.

**Comment by Cipscis** [comment]
Yeah that's fair. I assumed the TypeScript types implied that intent that it always be there, and so should get inherited, but the important thing is that the type definition matches the real data.
